### DISCUSSION TITLE
Network Requirements for Object Storage

### DISCUSSION THREAD

**Discussion by elpsycongroo** [milvus-io/milvus#36458] [Q&A and General discussion]
We are trying to deploy S3 outside the k8s cluster and I have some question about milvus network requirments for s3:

Does Milvus have bandwidth requirements for object storage? 
Is the read and write frequency for object storage high? 
Is the QPS related to the amount of stored data? Could you provide a reference value?

**Comment by yhmo** [comment]
It depends on runtime ingestion.
In my opinion, typically it is 2X ~ 3X the size of the data you inserted.

**Comment by xiaofan-luan** [comment]
> We are trying to deploy S3 outside the k8s cluster and I have some question about milvus network requirments for s3:
> 
> Does Milvus have bandwidth requirements for object storage? Is the read and write frequency for object storage high? Is the QPS related to the amount of stored data? Could you provide a reference value?

We don't read from S3 except for disaster recovery.
However the S3 iops is related to failure recovery speed. When failure happens, milvus need to load all the data back from S3 to local disk or in memory. So you have you have many segments/files, the outside S3 needs at least thousand QPS to suppor failure recovery in less than 1 minutes

**Reply by elpsycongroo** [reply]
Aside from failure recovery and loading collections, what other situations cause Milvus to read from S3?

**Reply by yhmo** [reply]
So far as I know, these actions will read S3:
- Query node loads collection
- Data node compaction action
- Index node builds index for segment

These actions will write S3:
- Data node persists new segments(generated by insert/upsert/compaction)
- Index node persists indexes

**Reply by elpsycongroo** [reply]
According to my understanding，after loading a collection, Milvus queries data from the local disk or memory. 
If querying only from memory, does loading large-scale collections put pressure on memory, and timely release necessary?
If querying from local disk, what's the requirement for the IOPS of the disk, ssd is suggested?

**Reply by yhmo** [reply]
For in-memory index, once the loading process is done, all the index data is in memory, and search request is executed in the memory.

For mmap enabled or DISKANN index, it might read data from disk. 
https://milvus.io/docs/disk_index.md
**To use DiskANN, note that
The Milvus data path should be mounted to an NVMe SSD for full performance**

https://milvus.io/docs/mmap.md#MMap-enabled-Data-Storage
**- What kind of local storage is required for memory mapping?
A high-quality disk enhances performance, with NVMe drives being the preferred option.**

We don't have a strong requirement for IOPS value for mmap and DISKANN. Low IOPS will affect the performance of mmap/diskann.

For the prerequisite of milvus, we mentioned 500 IOPS, which is mainly for the Etcd service.
https://milvus.io/docs/prerequisite-docker.md#Additional-disk-requirements
**Disk performance is critical to etcd. It is highly recommended that you use local NVMe SSDs. Slower disk response may cause frequent cluster elections that will eventually degrade the etcd service. Ideally, your disk should reach over 500 IOPS and below 10ms for the 99th percentile fsync latency. Read the etcd [Docs](https://etcd.io/docs/v3.5/op-guide/hardware/#disks) for more detailed requirements.**
